tests/spread/store/registries/task.yaml

```
@@ -0,0 +1,44 @@
+summary: test the registries commands
+
+environment:
+  SNAPCRAFT_ASSERTION_KEY: "$(HOST: echo ${SNAPCRAFT_ASSERTION_KEY})"
+  SNAPCRAFT_STORE_CREDENTIALS: "$(HOST: echo ${SNAPCRAFT_STORE_CREDENTIALS_STAGING})"
+
+prepare: |
+  if [[ -z "$SNAPCRAFT_STORE_CREDENTIALS" ]]; then
+    ERROR "No credentials set in env SNAPCRAFT_STORE_CREDENTIALS"
+  fi
+
+  if [[ -z "$SNAPCRAFT_ASSERTION_KEY" ]]; then
+    ERROR "No gpg key set in env SNAPCRAFT_ASSERTION_KEY"
+  fi
+
+  # setup snap gpg dir
+  mkdir -p "$HOME/.snap/gnupg"
+  chmod 700 "$HOME/.snap/gnupg"
+
+  # import a registered key
+  echo "$SNAPCRAFT_ASSERTION_KEY" | base64 --decode > store-key.txt
+  gpg --homedir "$HOME/.snap/gnupg" --import store-key.txt
+  rm -f store-key.txt
```

It's from the top of my head, but it should work 
```suggestion
  gpg --homedir "$HOME/.snap/gnupg" --import <(echo "$SNAPCRAFT_ASSERTION_KEY" | base64 --decode)
```

---

I tried a one-liner before and gpg was being weird about stdin.  I'll give this a try

---

I probably tried a pipe instead of a redirect